dc/s2n-quic-dc/src/stream/server/tokio/tcp/worker.rs

```diff
@@ -331,6 +341,10 @@ impl WorkerState {
                                 error: error.error,
                             };
                             continue;
+                        } else {
+                            // close the stream immediately and send a reset to the client
+                            let _ = socket.set_linger(Some(Duration::ZERO));
+                            drop(socket);
```

Comment:
Doesn't this hit the same close-without-sending / race RST and data packet issues?

Reply:
In this case we didn't actually send any data on the socket. If we didn't set linger=0, then the socket would be gracefully terminated with a FIN packet and the client would just get an EOF. It seems better to do a hard reset for errors like this so it at least bubbles up in logs on the client.